LiskHQ/lisk
# Reject blocks receipts with block slot/timestamp out of window - Closes #1191

### What was the problem?

Backporting fixes from 0.9.11 to 1.0.0 branch.

### How did I fix it?

Backported relevant code and tests.

### How to test it?

Run system functional tests file `test/functional/system/blocks/process/on_receive_block.js` or run `test/unit/modules/blocks/verify.js` unit tests.

### Review checklist

Check changes are backported correctly, review fixes added for rounds.
* The PR solves #1191
* All new code is covered with unit tests
* Linting passes
* Tests pass
* Commit messages follow the [commit guidelines](CONTRIBUTING.md#git-commit-messages)
* Documentation has been added/updated
